packages/ui-truncate-text/src/TruncateText/index.js

```diff
@@ -130,21 +130,34 @@ class TruncateText extends Component {
 
       this.truncate()
 
-      if (this.props.debounce === 0) {
-        this._resizeListener = addResizeListener(this._ref, this.update)
-      } else {
-        this._debounced = debounce(this.update, this.props.debounce, {
-          leading: true,
-          trailing: true
+      this._debounced = debounce(this.update, this.props.debounce, {
+        leading: true,
+        trailing: true
+      })
+
+      const { width: origWidth } = getBoundingClientRect(this._ref)
+      this._resizeListener = new ResizeObserver((entries) => {
+        // requestAnimationFrame call is needed becuase some truncatetext test cases
+        // failed due to ResizeObserver was not able to deliver all observations within a single animation frame
+        // see: https://stackoverflow.com/questions/49384120/resizeobserver-loop-limit-exceeded
```

Comment:
wont this cause some issue? In the SO discussion they mention something that this causes it to run in the next frame https://stackoverflow.com/a/58701523

Reply:
I haven't noticed any issues with this approach.
There is another way to solve this issue though:

```js
// ignore ResizeObserver loop limit exceeded
// this is ok in several scenarios according to 
// https://github.com/WICG/resize-observer/issues/38
before(() => {
  // called before any tests are run
  const e = window.onerror;
  window.onerror = function(err) {
    if(err === 'ResizeObserver loop limit exceeded') {
      console.warn('Ignored: ResizeObserver loop limit exceeded');
      return false;
    } else {
      return e(...arguments);
    }
  }
});
```

With this we we would simply ignore the error, I'm not sure this one is a better solution, what do you think?

Reply:
hmm I like the current one better, lets keep it